resources/user.py

```diff
@@ -89,16 +91,28 @@ def patch(self,user_id):
 
         if data['role']:
           user.role = RoleEnum(data['role'])
-        if (data['firstName'] != None):
+        if data['firstName'] is not None:
             user.firstName = data['firstName']
-        if (data['lastName'] != None):
+        if data['lastName'] is not None:
             user.lastName = data['lastName']
         if data['email']:
             user.email = data['email']
         if data['phone']:
             user.phone = data['phone']
-        if data['password']:
-            user.password = data['password']
+
+        # Reset Password
+        if data['current_password'] and data['new_password'] and data['confirm_password']:
+
+          # Step #1: Check if current password matches the one in the db
+          if not user.check_pw(data['current_password']):
+            return {"message": "Password does not match."}, 401
+
+          # Step #2: Check if new password and confirm password match
+          if data['new_password'] != data['confirm_password']:
+            return {"message": "New password does not match."}, 422
+
+          # Step #3: Set the new password
+          user.hash_digest = UserModel.hash_pw(data['new_password'])
```

Comment:
This can be:
```suggestion
          user.password = data['new_password']
```

It is hashed when the user is saved to db.